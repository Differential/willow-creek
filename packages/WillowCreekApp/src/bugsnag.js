import { Client } from 'bugsnag-react-native';
import Config from 'react-native-config';

const bugsnag = new Client(Config.BUGSNAG_API_KEY);

export default bugsnag;
