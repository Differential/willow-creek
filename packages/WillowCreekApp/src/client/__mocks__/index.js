import { MockedProvider } from 'react-apollo/test-utils';
import ApolloClient from 'apollo-client';
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { SchemaLink } from 'apollo-link-schema';
import { ApolloLink } from 'apollo-link';
import { testSchema as typeDefs } from 'apollos-church-api';

import cache from '../cache';
import clientStateLink from '../clientStateLink';

export default MockedProvider;

const schema = makeExecutableSchema({
  typeDefs,
  resolverValidationOptions: {
    requireResolversForResolveType: false,
  },
});
addMockFunctionsToSchema({ schema });

let link = new SchemaLink({ schema });
if (clientStateLink) {
  link = ApolloLink.from([clientStateLink, link]);
}

export const client = new ApolloClient({
  cache,
  link,
});
