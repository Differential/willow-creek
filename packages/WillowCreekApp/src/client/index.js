import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { ApolloLink } from 'apollo-link';
import SplashScreen from 'react-native-splash-screen';

import httpLink from './httpLink';
import clientStateLink from './clientStateLink';
import authLink from './authLink'; // eslint-disable-line
import cache, { ensureCacheHydration } from './cache';

const link = clientStateLink.concat(ApolloLink.from([authLink, httpLink]));

export const client = new ApolloClient({
  link,
  cache,
  queryDeduplication: true,
  shouldBatch: true,
});

class ClientProvider extends PureComponent {
  static propTypes = {
    client: PropTypes.shape({
      cache: PropTypes.shape({}),
    }),
  };

  static defaultProps = {
    client,
  };

  async componentDidMount() {
    try {
      await ensureCacheHydration;
    } catch (e) {
      throw e;
    } finally {
      if (SplashScreen && SplashScreen.hide) SplashScreen.hide();
    }
  }

  render() {
    return <ApolloProvider {...this.props} client={client} />;
  }
}

export default ClientProvider;
