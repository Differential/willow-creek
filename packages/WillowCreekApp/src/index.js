import React from 'react';
import { StatusBar } from 'react-native';
import { createStackNavigator } from 'react-navigation';
// import { Sentry } from 'react-native-sentry';

import { BackgroundView, withTheme } from '@apollosproject/ui-kit';
import MediaPlayer from 'WillowCreekApp/src/ui/MediaPlayer';
import Providers from './Providers';
import NavigationService from './NavigationService';
import ContentSingle from './content-single';
import Tabs from './tabs';
import Auth from './auth';
import PersonalDetails from './user-settings/PersonalDetails';
import ChangePassword from './user-settings/ChangePassword';

const AppStatusBar = withTheme(({ theme }) => ({
  barStyle: 'dark-content',
  backgroundColor: theme.colors.paper,
}))(StatusBar);

const AppNavigator = createStackNavigator(
  {
    Tabs,
    ContentSingle,
    Auth,
    PersonalDetails,
    ChangePassword,
  },
  {
    initialRouteName: 'Tabs',
    mode: 'modal',
    headerMode: 'screen',
  }
);

const App = () => (
  <Providers>
    <BackgroundView>
      <AppStatusBar barStyle="dark-content" />
      <AppNavigator
        ref={(navigatorRef) => {
          NavigationService.setTopLevelNavigator(navigatorRef);
        }}
      />
      <MediaPlayer />
    </BackgroundView>
  </Providers>
);

export default App;
