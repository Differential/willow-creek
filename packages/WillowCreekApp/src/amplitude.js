import amplitude from 'amplitude-js';
import Config from 'react-native-config';
import gql from 'graphql-tag';
import { get } from 'lodash';
import { getVersion } from 'react-native-device-info';
import { client } from './client';

amplitude.getInstance().init(Config.AMPLITUDE_KEY, null, {
  useNativeDeviceInfo: true,
});

export const track = ({ eventName, properties = null }) => {
  amplitude
    .getInstance()
    .logEvent(eventName, properties, (...args) => console.log(args));
};

export const identify = () => {
  const data = client.readQuery({
    query: gql`
      query currentCampusId {
        currentUser {
          id
          profile {
            id
            campus {
              id
              name
            }
          }
        }
      }
    `,
  });

  amplitude.getInstance().setUserId(get(data, 'currentUser.profile.id'));

  amplitude.getInstance().setUserProperties({
    campusName: get(data, 'currentUser.profile.campus.name'),
    appVersion: getVersion(),
  });
};
