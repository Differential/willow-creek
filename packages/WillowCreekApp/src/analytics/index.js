import gql from 'graphql-tag';
import DeviceInfo from 'react-native-device-info';
import { Platform } from 'react-native';

import { client } from '../client';

const anonymousId = DeviceInfo.getUniqueID();

const deviceInfo = {
  platform: Platform.OS === 'ios' ? 'iOS' : 'Android',
  deviceId: anonymousId,
  deviceModel: DeviceInfo.getModel(),
  osVersion: DeviceInfo.getSystemVersion(),
  appVersion: DeviceInfo.getVersion(),
};

const trackMutation = gql`
  mutation track($input: AnalyticsTrackInput!) {
    trackEvent(input: $input) {
      success
    }
  }
`;

const identifyMutation = gql`
  mutation identify($input: AnalyticsIdentifyInput!) {
    identifySelf(input: $input) {
      success
    }
  }
`;

const propertiesToGqlInput = (props = []) =>
  Object.keys(props).map((key) => ({
    field: key,
    value: props[key],
  }));

export const track = ({ eventName, properties }) =>
  client.mutate({
    mutation: trackMutation,
    variables: {
      input: {
        anonymousId,
        deviceInfo,
        eventName,
        properties: propertiesToGqlInput(properties),
      },
    },
  });

export const identify = () =>
  client.mutate({
    mutation: identifyMutation,
    variables: {
      input: {
        anonymousId,
        deviceInfo,
      },
    },
  });

export const events = {
  LikeContent: 'Like Content',
  UnlikeContent: 'Unlike Content',
  ViewContent: 'View Content',
  ShareContent: 'Share Content',
  UserLogin: 'User Login',
  UserSignup: 'User Signup',
  UserLogout: 'UserLogout',
  UserForgotPassword: 'User Forgot Password',
  UserPlayedMedia: 'User Played Media',
};

export default {
  track,
  identify,
  events,
};
