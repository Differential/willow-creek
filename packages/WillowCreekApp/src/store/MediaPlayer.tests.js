import { client } from 'WillowCreekApp/src/client';
import gql from 'graphql-tag';

import { resolvers, defaults } from '.';

const query = gql`
  query {
    mediaPlayer @client {
      currentTrack {
        mediaSourceUri
        posterSourceUri
        title
        artist
      }
      isPlaying
      isFullscreen
      isVisible
    }
  }
`;

const testTrack = {
  parentId: 'parentId',
  mediaSourceUri: 'mediaSourceUri',
  posterSourceUri: 'posterSourceUri',
  title: 'title',
  artist: 'artist',
};

describe('MediaPlayer Store', () => {
  beforeEach(() => {
    client.writeData({ data: defaults });
  });

  it('plays a track', async () => {
    resolvers.Mutation.mediaPlayerPlayNow({}, testTrack, {
      cache: client.cache,
    });
    expect(client.query({ query })).resolves.toMatchSnapshot();
  });
  it('updates player state', async () => {
    resolvers.Mutation.mediaPlayerPlayNow({}, testTrack, {
      cache: client.cache,
    });
    expect(client.query({ query })).resolves.toMatchSnapshot();
    resolvers.Mutation.mediaPlayerUpdateState(
      {},
      { isPlaying: false },
      { cache: client.cache }
    );
    expect(client.query({ query })).resolves.toMatchSnapshot();
    resolvers.Mutation.mediaPlayerUpdateState(
      {},
      { isPlaying: true },
      { cache: client.cache }
    );
  });
});
