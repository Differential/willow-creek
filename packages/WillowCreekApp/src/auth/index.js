import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View, SafeAreaView } from 'react-native';

import {
  FlexedView,
  TabView,
  TabSceneMap as SceneMap,
  H1,
  H5,
  styled,
  Icon,
  ButtonLink,
  withTheme,
} from '@apollosproject/ui-kit';

import { track } from 'WillowCreekApp/src/analytics';

import LoginForm from './login';
import SignUpForm from './signup';

export LoginButton from './LoginButton';
export ProtectedAction from './ProtectedAction';

const Title = styled(({ theme }) => ({ color: theme.colors.primary }))(H1);

const BrandIcon = withTheme(({ theme }) => ({
  name: 'brand-icon',
  size: theme.sizing.baseUnit * 2.25,
  marginVertical: theme.sizing.baseUnit,
  fill: theme.colors.primary,
}))(Icon);

const HeaderContainer = styled(({ theme }) => ({
  backgroundColor: theme.colors.background.paper,
}))(SafeAreaView);

const Header = styled(({ theme }) => ({
  padding: theme.sizing.baseUnit,
  paddingBottom: theme.sizing.baseUnit * 1.5,
}))(View);

const CancelButton = styled(({ theme }) => ({
  alignSelf: 'flex-end',
  paddingTop: theme.sizing.baseUnit * 0.75,
  paddingRight: theme.sizing.baseUnit,
}))(ButtonLink);

class Auth extends PureComponent {
  static navigationOptions = {
    header: null,
  };

  tabRoutes = [
    { title: 'Sign In', key: 'login' },
    { title: 'Register', key: 'signup' },
  ];

  static propTypes = {
    navigation: PropTypes.shape({
      goBack: PropTypes.func,
    }),
  };

  handleFinish = () => {
    // trigger the auth modal to close
    track({ eventName: 'UserLogin' });
    this.props.navigation.goBack();
  };

  renderLogin = () => <LoginForm onLogin={this.handleFinish} />;

  renderSignup = () => <SignUpForm onSignup={this.handleFinish} />;

  render() {
    return (
      <FlexedView>
        <HeaderContainer>
          <CancelButton onPress={this.handleFinish}>Cancel</CancelButton>
          <Header>
            <BrandIcon />
            <Title>Welcome!</Title>
            <H5>Please sign in to continue</H5>
          </Header>
        </HeaderContainer>
        <TabView
          routes={this.tabRoutes}
          renderScene={SceneMap({
            login: this.renderLogin,
            signup: this.renderSignup,
          })}
        />
      </FlexedView>
    );
  }
}

export default Auth;
