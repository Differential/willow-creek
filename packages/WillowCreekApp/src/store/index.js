import { merge, get } from 'lodash';
import gql from 'graphql-tag';
import getLoginState from 'WillowCreekApp/src/auth/getLoginState';
import { track, events } from 'WillowCreekApp/src/analytics';
import { client } from '../client'; // eslint-disable-line
import getAuthToken from './getAuthToken';
// TODO: this will require more organization...ie...not keeping everything in one file.
// But this is simple while our needs our small.

export const schema = `
  type Query {
    authToken: String
    mediaPlayer: MediaPlayerState
    isLoggedIn: Boolean
  }

  type Mutation {
    logout
    mediaPlayerUpdateState(isPlaying: Boolean, isFullscreen: Boolean, isVisible: Boolean): Boolean
    mediaPlayerSetPlayhead(currentTime: Float): Boolean
    mediaPlayerPlayNow(
      parentId: ID,
      mediaSource: VideoMediaSource!,
      posterSources: [ImageMediaSource],
      title: String,
      artist: String,
      isVideo: Boolean,
    ): Boolean

    handleLogin(authToken: String!)
  }

  type MediaPlayerState {
    currentTrack: MediaPlayerTrack
    isPlaying: Boolean
    isFullscreen: Boolean
    isVisible: Boolean
    currentTime: Float
  }

  type MediaPlayerProgress {
    currentTime: Float
    playableDuration: Float
    seekableDuration: Float
    duration: Float
  }

  type MediaPlayerTrack {
    id: ID!
    parentId: ID
    mediaSource: VideoMediaSource!
    posterSources: [ImageMediaSource]
    title: String
    artist: String
    isVideo: Boolean
  }
`;

export const defaults = {
  __typename: 'Query',
  authToken: null,
  mediaPlayer: {
    __typename: 'MediaPlayerState',
    currentTrack: null,
    isPlaying: false,
    isFullscreen: false,
    isVisible: false,
    currentTime: 0,
    showVideo: true,
    muted: false,
  },
};

let trackId = 0;

export const resolvers = {
  Query: {
    isLoggedIn: () => {
      // When logging out, this query returns an error.
      // Rescue the error, and return false.
      try {
        const { authToken } = client.readQuery({ query: getAuthToken });
        return !!authToken;
      } catch (e) {
        return false;
      }
    },
  },
  Mutation: {
    logout: (root, variables, { cache }) => {
      client.resetStore();
      cache.writeData({
        data: defaults,
      });
      track({ eventName: events.UserLogout });
      return null;
    },

    handleLogin: async (root, { authToken }, { cache }) => {
      try {
        await cache.writeQuery({
          query: getAuthToken,
          data: { authToken },
        });
        await cache.writeQuery({
          query: getLoginState,
          data: { isLoggedIn: true },
        });
        await cache.writeData({
          data: { authToken },
        });

        track({ eventName: events.UserLogin });
      } catch (e) {
        throw e.message;
      }

      return null;
    },

    mediaPlayerPlayNow: (root, trackInfo, { cache }) => {
      const query = gql`
        query {
          mediaPlayer {
            isFullscreen
          }
        }
      `;
      const mediaTrack = merge(
        {
          __typename: 'MediaPlayerTrack',
          parentId: null,
          mediaSource: null,
          posterSources: null,
          title: null,
          artist: null,
          isVideo: false,
        },
        trackInfo
      );

      const { mediaPlayer } = cache.readQuery({ query });

      const newMediaPlayerState = {
        __typename: 'MediaPlayerState',
        isPlaying: true,
        isVisible: true,
        isFullscreen: mediaTrack.isVideo
          ? true
          : (mediaPlayer && mediaPlayer.isFullscreen) || false,
        currentTrack: mediaTrack,
        currentTime: 0,
        showVideo: mediaTrack.isVideo,
        muted: false,
      };

      if (
        // if same track
        mediaPlayer &&
        mediaPlayer.currentTrack &&
        mediaPlayer.currentTrack.mediaSource.uri === mediaTrack.mediaSource.uri
      ) {
        // use the same Id
        newMediaPlayerState.currentTrack.id = mediaPlayer.currentTrack.id;
      } else {
        // otherwise reset progress and use new Id
        newMediaPlayerState.currentTrack.id = trackId;
        newMediaPlayerState.progress = null;
        trackId += 1;
      }

      cache.writeData({
        query,
        data: {
          mediaPlayer: newMediaPlayerState,
        },
      });

      track({
        eventName: events.UserPlayedMedia,
        properties: {
          uri: mediaTrack.uri,
          title: mediaTrack.title,
          type: mediaTrack.isVideo ? 'Video' : 'Audio',
        },
      });
      return null;
    },
    mediaPlayerUpdateState: (
      root,
      { isPlaying, isFullscreen, isVisible, showVideo, muted },
      { cache }
    ) => {
      const query = gql`
        query {
          mediaPlayer @client {
            isPlaying
            isFullscreen
            isVisible
            showVideo
            muted
          }
        }
      `;
      const { mediaPlayer } = cache.readQuery({ query });
      cache.writeQuery({
        query,
        data: {
          mediaPlayer: merge(mediaPlayer, {
            isPlaying,
            isFullscreen,
            isVisible,
            showVideo,
            muted,
            __typename: 'MediaPlayerState',
          }),
        },
      });
      return null;
    },
    mediaPlayerSetPlayhead: (root, { currentTime }, { cache }) => {
      const query = gql`
        query {
          mediaPlayer @client {
            currentTime
          }
        }
      `;
      const { mediaPlayer } = cache.readQuery({ query });
      cache.writeQuery({
        query,
        data: {
          mediaPlayer: {
            __typename: 'MediaPlayerState',
            currentTime:
              currentTime || get(mediaPlayer.progress, 'currentTime') || 0,
          },
        },
      });
      return null;
    },
  },
};
