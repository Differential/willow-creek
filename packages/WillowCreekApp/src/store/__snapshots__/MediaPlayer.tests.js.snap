// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPlayer Store plays a track 1`] = `
Object {
  "data": Object {
    "mediaPlayer": Object {
      "__typename": "MediaPlayerState",
      "currentTrack": Object {
        "__typename": "MediaPlayerTrack",
        "artist": "artist",
        "mediaSourceUri": "mediaSourceUri",
        "posterSourceUri": "posterSourceUri",
        "title": "title",
      },
      "isFullscreen": false,
      "isPlaying": true,
      "isVisible": true,
    },
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`MediaPlayer Store updates player state 1`] = `
Object {
  "data": Object {
    "mediaPlayer": Object {
      "__typename": "MediaPlayerState",
      "currentTrack": Object {
        "__typename": "MediaPlayerTrack",
        "artist": "artist",
        "mediaSourceUri": "mediaSourceUri",
        "posterSourceUri": "posterSourceUri",
        "title": "title",
      },
      "isFullscreen": false,
      "isPlaying": true,
      "isVisible": true,
    },
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`MediaPlayer Store updates player state 2`] = `
Object {
  "data": Object {
    "mediaPlayer": Object {
      "__typename": "MediaPlayerState",
      "currentTrack": Object {
        "__typename": "MediaPlayerTrack",
        "artist": "artist",
        "mediaSourceUri": "mediaSourceUri",
        "posterSourceUri": "posterSourceUri",
        "title": "title",
      },
      "isFullscreen": false,
      "isPlaying": false,
      "isVisible": true,
    },
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;
